src/test/credentials/provider/ssoAccessTokenProvider.test.ts

```diff
@@ -0,0 +1,230 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import * as AWS from 'aws-sdk'
+import * as sinon from 'sinon'
+import { SSOOIDC } from 'aws-sdk'
+import { DiskCache } from '../../../credentials/sso/diskCache'
+import { SsoAccessTokenProvider } from '../../../credentials/sso/ssoAccessTokenProvider'
+import { CreateTokenResponse, StartDeviceAuthorizationResponse } from 'aws-sdk/clients/ssooidc'
+import { SsoClientRegistration } from '../../../credentials/sso/ssoClientRegistration'
+import { SsoAccessToken } from '../../../credentials/sso/ssoAccessToken'
+
+describe('SsoAccessTokenProvider', () => {
+    let sandbox = sinon.createSandbox()
+
+    const ssoRegion = 'fakeRegion'
+    const ssoUrl = 'fakeUrl'
+    const ssoOidcClient = new SSOOIDC({ region: 'us-west-2' })
+    const cache = new DiskCache()
+    const sut = new SsoAccessTokenProvider(ssoRegion, ssoUrl, ssoOidcClient, cache)
+
+    const HOUR_IN_MS = 3600000
+
+    const validAccessToken = {
+        startUrl: ssoUrl,
+        region: ssoRegion,
+        accessToken: 'dummyAccessToken',
+        expiresAt: new Date(Date.now() + HOUR_IN_MS).toISOString(),
+    }
+
+    const fakeCreateTokenResponse: SSOOIDC.CreateTokenResponse = {
+        accessToken: 'dummyAccessToken',
+        expiresIn: 120,
+    }
+
+    const validRegistation: SsoClientRegistration = {
+        clientId: 'aString',
+        clientSecret: 'aString',
+        expiresAt: new Date(Date.now() + HOUR_IN_MS).toISOString(),
+    }
+
+    const validAuthorization: StartDeviceAuthorizationResponse = {
+        expiresIn: 120,
+        deviceCode: 'dummyCode',
+        userCode: 'dummyUserCode',
+        verificationUri: 'aUrl',
+        verificationUriComplete: 'aUrlComplete',
+        interval: 1,
+    }
+
+    function setUpStubCache(returnRegistration?: SsoClientRegistration, returnAccessToken?: SsoAccessToken) {
```

Comment:
`returnAccessToken` is never set in any calls to `setUpStubCache`. Remove it (and the if statement that uses it) or use it; if you do use it and you only want to set one of `returnRegistration` or `returnAccessToken`, encapsulate the params in a single object.

Reply:
Great advice, thanks.  For this case, I will not need the second param.